### ISSUE TITLE
COM Exception occurred

### ISSUE THREAD

**Issue by mpking828** [phw198/OutlookGoogleCalendarSync#247]
Windows 10
Outlook 2016
McAfee AV

I have OGCS set to start with windows startup.  It will throw a COM error on every startup now.  I just now (i know, belatedly) realized I should crank up the logging level, and I'm doing it as we speak.  The logs I have are pretty useless (Last entry was from yesterday).

When I manually start, everything is fine.

I do have this from the windows App log:

Event 1026
Application: OutlookGoogleCalendarSync.exe
Framework Version: v4.0.30319
Description: The process was terminated due to an unhandled exception.
Exception Info: System.NullReferenceException
   at OutlookGoogleCalendarSync.MainForm.Dispose(Boolean)
   at System.ComponentModel.Component.Finalize()

followed by Event 1000
Faulting application name: OutlookGoogleCalendarSync.exe, version: 2.5.0.0, time stamp: 0x5932ed85
Faulting module name: unknown, version: 0.0.0.0, time stamp: 0x00000000
Exception code: 0xc0000005
Fault offset: 0x0a801ca9
Faulting process id: 0x7fbc
Faulting application start time: 0x01d2ea8ee9ce139b
Faulting application path: C:\Users\310089811\AppData\Local\Apps\2.0\3KNYQXQY.0VB\ZD34KTXN.9N2\outl..tion_a30846ba3587a523_0002.0005_3590f07bd0c34f4b\OutlookGoogleCalendarSync.exe
Faulting module path: unknown
Report Id: 793a44a7-90fe-4ef3-92e5-15a091f1bb60
Faulting package full name: 
Faulting package-relative application ID: 

I'll switch the logging level, reboot, and report back.

**Comment by hmonteiro** [comment]
I have the same problem and here is the log:
`2017-06-21 15:24:27,336 DEBUG 1 OutlookGoogleCalendarSync.XMLManager [0] - Retrieved setting 'Portable' with value 'false'
2017-06-21 15:24:27,338 DEBUG 1 OutlookGoogleCalendarSync.XMLManager [0] - Retrieved setting 'LoggingLevel' with value 'DEBUG'
2017-06-21 15:24:27,338 INFO 1 OutlookGoogleCalendarSync.Settings [0] - Logging level configured to 'DEBUG'
2017-06-21 15:24:27,339 INFO 1 OutlookGoogleCalendarSync.Program [0] - Purging log files older than 30 days...
2017-06-21 15:24:27,342 INFO 1 OutlookGoogleCalendarSync.Program [0] - Purge complete.
2017-06-21 15:24:27,345 DEBUG 1 OutlookGoogleCalendarSync.Program [0] - Loading settings from file.
2017-06-21 15:24:27,508 DEBUG 3 OutlookGoogleCalendarSync.XMLManager [0] - Retrieved setting 'CompletedSyncs' with value '1893'
2017-06-21 15:24:27,510 DEBUG 3 OutlookGoogleCalendarSync.XMLManager [0] - Retrieved setting 'Donor' with value 'false'
2017-06-21 15:24:27,511 DEBUG 3 OutlookGoogleCalendarSync.XMLManager [0] - Retrieved setting 'Subscribed' with value '2000-01-01T00:00:00'
2017-06-21 15:24:27,512 DEBUG 3 OutlookGoogleCalendarSync.XMLManager [0] - Retrieved setting 'HideSplashScreen' with value 'false'
2017-06-21 15:24:27,512 DEBUG 3 OutlookGoogleCalendarSync.Splash [0] - Showing splash screen.
2017-06-21 15:24:27,535 DEBUG 1 OutlookGoogleCalendarSync.Updater [0] - Automatic update check requested.
2017-06-21 15:24:27,538 INFO 1 OutlookGoogleCalendarSync.SettingsProxy [0] - Using default proxy (app.config / IE).
2017-06-21 15:24:27,538 INFO 1 OutlookGoogleCalendarSync.SettingsProxy [0] - Setting system-wide proxy.
2017-06-21 15:24:27,563 DEBUG 1 OutlookGoogleCalendarSync.SettingsProxy [0] - Testing the system proxy.
2017-06-21 15:24:27,568 DEBUG 1 OutlookGoogleCalendarSync.SettingsProxy [0] - Confirmation of configured proxy: http://www.google.com
2017-06-21 15:24:27,636 INFO 1 OutlookGoogleCalendarSync.Updater [0] - This is a Squirrel aware ClickOnce install.
2017-06-21 15:24:27,637 DEBUG 1 OutlookGoogleCalendarSync.Updater [0] - Checking for Squirrel update...
2017-06-21 15:24:27,684 WARN 1 OutlookGoogleCalendarSync.NotificationTray [0] - Custom TZDB source failed. Falling back to NodaTime.dll
2017-06-21 15:24:27,706 INFO 1 OutlookGoogleCalendarSync.NotificationTray [0] - Using NodaTime TZDB: 2017b (mapping: 13050)
2017-06-21 15:24:27,715 DEBUG 7 OutlookGoogleCalendarSync.NotificationTray [0] - Checking for new timezone database...
2017-06-21 15:24:27,716 DEBUG 1 OutlookGoogleCalendarSync.MainForm [0] - Initialiasing MainForm.
2017-06-21 15:24:27,808 INFO 1 OutlookGoogleCalendarSync.OutlookCalendar [0] - Attaching to the already running Outlook process.
2017-06-21 15:24:27,927 ERROR 1 OutlookGoogleCalendarSync.OGCSexception [0] - System.InvalidCastException: Unable to cast COM object of type 'Microsoft.Office.Interop.Outlook.ApplicationClass' to interface type 'Microsoft.Office.Interop.Outlook._Application'. This operation failed because the QueryInterface call on the COM component for the interface with IID '{00063001-0000-0000-C000-000000000046}' failed due to the following error: Library not registered. (Exception from HRESULT: 0x8002801D (TYPE_E_LIBNOTREGISTERED)).
2017-06-21 15:24:27,927 ERROR 1 OutlookGoogleCalendarSync.OGCSexception [0] - Code: 0x80004002;-2147467262
2017-06-21 15:24:27,928 FATAL 1 OutlookGoogleCalendarSync.Program [0] - Application unexpectedly terminated!
2017-06-21 15:24:28,077 DEBUG 7 OutlookGoogleCalendarSync.NotificationTray [0] - Already have latest TZDB version.
2017-06-21 15:24:31,880 ERROR 6 OutlookGoogleCalendarSync.OGCSexception [0] - System.Exception: Remote release File is empty or corrupted
2017-06-21 15:24:31,927 ERROR 6 OutlookGoogleCalendarSync.OGCSexception [0] - Code: 0x80131500;-2146233088
2017-06-21 15:24:31,927 ERROR 6 OutlookGoogleCalendarSync.OGCSexception [0] -    at Squirrel.UpdateManager.CheckForUpdateImpl.<CheckForUpdate>d__2.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.UpdateManager.<CheckForUpdate>d__7.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at OutlookGoogleCalendarSync.Updater.<githubCheck>d__5.MoveNext()
2017-06-21 15:24:32,270 WARN 1 OutlookGoogleCalendarSync.Program [0] - Tidying down any remaining Outlook references, as OGCS crashed out.
2017-06-21 15:24:32,285 INFO 1 OutlookGoogleCalendarSync.OutlookCalendar [0] - Attaching to the already running Outlook process.
2017-06-21 15:24:32,285 ERROR 1 OutlookGoogleCalendarSync.OGCSexception [0] - System.InvalidCastException: Unable to cast COM object of type 'Microsoft.Office.Interop.Outlook.ApplicationClass' to interface type 'Microsoft.Office.Interop.Outlook._Application'. This operation failed because the QueryInterface call on the COM component for the interface with IID '{00063001-0000-0000-C000-000000000046}' failed due to the following error: Library not registered. (Exception from HRESULT: 0x8002801D (TYPE_E_LIBNOTREGISTERED)).
2017-06-21 15:24:32,285 ERROR 1 OutlookGoogleCalendarSync.OGCSexception [0] - Code: 0x80004002;-2147467262
2017-06-21 15:24:32,285 INFO 1 OutlookGoogleCalendarSync.OutlookCalendar [0] - It appears Outlook has been restarted after OGCS was started. Reconnecting...
2017-06-21 15:24:32,285 INFO 1 OutlookGoogleCalendarSync.OutlookCalendar [0] - Attaching to the already running Outlook process.
2017-06-21 15:24:32,353 INFO 1 OutlookGoogleCalendarSync.Program [0] - Application closed.
`

**Comment by mpking828** [comment]
[OGcalsync.txt](https://github.com/phw198/OutlookGoogleCalendarSync/files/1092377/OGcalsync.txt)

So I set the logging level to 'ALL', but it doesn't appear to write a log file when it crashes.

Here is my OGcalsysnc.log, but it's dated 9:21AM, yet I crashed at 12:51pm.

I did get this in the windows event log at the same time:

The server {0006F03A-0000-0000-C000-000000000046} did not register with DCOM within the required timeout.
